### DISCUSSION TITLE
Text Alignment

### DISCUSSION THREAD

**Discussion by lasmith** [mdx-editor/editor#207] [Ideas]
Is there any functionality to align text (left / center / right). I cannot see any plugins that enable this functionality?

Assuming its not implemented, how would one go about adding this functionality to the toolbar?

Thanks

Laurence

**Comment by petyosi** [comment]
How are you doing text alignment in markdown?

**Reply by lasmith** [reply]
We are not currently but its a feature request from the users of a product we are developing that embeds the mdxeditor.

I appreciate markdown does not support it but the general approach seems to be to use html tags (e.g. text-align). The mdxeditor already does something similar with underline (which is not in the markdown standard). Currently that puts a "u" tag in (which TBH is a bit out of date but works..). So just wondering if this is possible?

If not how would one go about implementing it (new toolbar button, hook to insert a "p" or "div" tag with a CSS style applied to center the text.

**Reply by petyosi** [reply]
The editor has a basic [generic HTML support](https://mdxeditor.dev/editor/docs/html-support). I've done a few (very raw) examples of basic tag management [in this example](https://github.com/mdx-editor/editor/blob/main/src/examples/html.tsx). Notice that you need certain familiarity with the Lexical framework for the more complex stuff.

**Reply by lasmith** [reply]
Thanks seems achievable via that example - just what I was after. 

Might be nice as a pre-built plugin at some point, along with changing of text colour :)
